api/task_manager.cc

```diff
@@ -25,18 +26,18 @@ namespace tm = httpd::task_manager_json;
 using namespace json;
 using namespace seastar::httpd;
 
-tm::task_status make_status(tasks::task_status status) {
+tm::task_status make_status(tasks::task_status status, sharded<service::storage_service>& ss) {
     auto start_time = db_clock::to_time_t(status.start_time);
     auto end_time = db_clock::to_time_t(status.end_time);
     ::tm st, et;
     ::gmtime_r(&end_time, &et);
     ::gmtime_r(&start_time, &st);
 
     std::vector<tm::task_identity> tis{status.children.size()};
-    std::ranges::transform(status.children, tis.begin(), [] (const auto& child) {
+    std::ranges::transform(status.children, tis.begin(), [&ss] (const auto& child) {
         tm::task_identity ident;
         ident.task_id = child.task_id.to_sstring();
-        ident.node = fmt::format("{}", child.node);
+        ident.node = fmt::format("{}", ss.local_is_initialized() ? ss.local().host2ip(child.host_id) : gms::inet_address{});
```

Comment:
Better pass gossiper to the API handler instead of storage_service. Storage service is not a primary source of the mapping, gossiper is. `ss::host2ip` should not exist.